app/src/main/java/com/imdb_compose/MainActivity.kt

```diff
@@ -1,47 +1,1195 @@
+            .aspectRatio(aspectRatio, true)
+            .clip(
+                if (clip) {
+                    RoundedCornerShape(16.dp)
+                } else {
+                    RoundedCornerShape(0.dp)
+                }
+            ),
+        contentScale = if (clip) {
+            ContentScale.Crop
+        } else {
+            ContentScale.FillBounds
+        },
+        contentDescription = contentDescription,
+        loading = { isLoading() }
     )
 }
 
-@Preview(showBackground = true)
 @Composable
-fun GreetingPreview() {
-    Imdb_composeTheme {
-        Greeting("Android")
+fun Tags(txt: String) {
+    Box(
+        modifier = Modifier
+            .shadow(
+                color = gray600,
+                offsetX = 0.dp,
+                offsetY = 0.dp,
+                blurRadius = 2.dp,
+                blurRadiusFilter = "NORMAL"
+            )
+    ) {
+        Box(
+            modifier = Modifier
+                .shadow(
+                    color = gray300,
+                    offsetX = 0.dp,
+                    offsetY = 0.dp,
+                    blurRadius = 2.dp,
+                    blurRadiusFilter = "INNER"
+                )
+        ) {
+            Text(
+                modifier = Modifier.padding(all = 8.dp),
+                text = txt,
+                color = Color.White
+            )
+        }
     }
 }
+
+@Composable
+fun Pager(images: State<Images?>) {
+    val pagerState = rememberPagerState { images.value?.backdrops?.size!! }
+
+    Row (modifier = Modifier
+        .fillMaxWidth()
+        .padding(bottom = 16.dp)) {
+        Column {
+            HorizontalPager(
+                state = pagerState,
+                modifier = Modifier.fillMaxWidth()
+            ) { i ->
+                Card(modifier = Modifier.fillMaxWidth()) {
+                    ImageAsync(
+                        backDropPath = when (i) {
+                            0 -> images.value?.backdrops!![0].file_path
+                            1 -> images.value?.backdrops!![1].file_path
+                            2 -> images.value?.backdrops!![2].file_path
+                            3 -> images.value?.backdrops!![3].file_path
+                            else -> images.value?.backdrops!![4].file_path
+                        },
+                        aspectRatio = when (i) {
+                            0 -> images.value?.backdrops!![0].aspect_ratio
+                            1 -> images.value?.backdrops!![1].aspect_ratio
+                            2 -> images.value?.backdrops!![2].aspect_ratio
+                            3 -> images.value?.backdrops!![3].aspect_ratio
+                            else -> images.value?.backdrops!![4].aspect_ratio
+                        },
+                        contentDescription = ""
+                    )
+                }
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun Carosuel(images: State<Images?>) {
```

Comment:
What's the difference between this function and Pager? The code looks similar. Is it a case where the functionality is the same, but the UI is slightly different? Would it be possible to parameterize the difference into the `modifier` argument?

Reply:
it's not being used, i made it first before i knew there was a horizontalPager. I thought the horizontalPager was a carousel.